content/doc/developer/publishing/releasing-cd.adoc

```diff
@@ -104,37 +134,48 @@ updates:
      </properties>
 ----
 +
-Thus your resulting `.mvn/maven.config` should look like:
-+
-[source]
-----
--Pconsume-incrementals
--Pmight-produce-incrementals
--Dchangelist.format=%d.v%s
-----
-+
-and your `pom.xml` should contain a top-level
+In this typical case, a CI/release build (`-Dset.changelist` specified) will be of the form `123.vabcdef456789`.
+A snapshot build will be `999999-SNAPSHOT`: arbitrary but treated as a snapshot by Maven and newer than any release.
 +
-[source,xml]
-----
-<version>${changelist}</version>
-----
+IMPORTANT: Note that you will very quickly create releases with version numbers greater than 100, as the major version component corresponds to the number of commits in the branch you're releasing from.
+If you're not ready to commit to all future versions of your plugin being this large, see the next option.
 +
-as well as a `<properties>` section containing
+NOTE: It is worth communicating this to your users, as they will see a very different version number format than before.
+The best way to do this is to add a line to the release notes: link:https://github.com/jenkinsci/azure-artifact-manager-plugin/releases/tag/86.va2aa4b1038c7[example note].
+
+Manually controlled prefix::
+If you do not want to have (major) version numbers increase quickly, like with fully automated versioning described above, keep `<revision>` in the `<properties>` section, setting it to the prefix (`major`, `major.minor`, etc., depending on how much of the version number you want to manually manage) and use it as part of the top-level `<version>` element:
 +
-[source,xml]
+[source,diff]
 ----
-<changelist>999999-SNAPSHOT</changelist>
+--- a/.mvn/maven.config
++++ b/.mvn/maven.config
+@@ -1,2 +1,3 @@
+ -Pconsume-incrementals
+ -Pmight-produce-incrementals
++-Dchangelist.format=%d.v%s
+--- a/pom.xml
++++ b/pom.xml
+@@ -10,12 +10,12 @@
+   <artifactId>some-library-wrapper</artifactId>
+-  <version>${revision}${changelist}</version>
++  <version>${revision}.${changelist}</version>
+   <packaging>hpi</packaging>
+   <properties>
+-    <revision>1.2.3</revision>
+-    <changelist>-SNAPSHOT</changelist>
++    <revision>1</revision>
++    <changelist>999999-SNAPSHOT</changelist>
+     <jenkins.version>2.176.4</jenkins.version>
 ----
```

Comment:
Why not directly a suggestion? I do not see the point of a diff, it could not match with the plugin's config. It is more clear and straigforward.

.mvn/maven.config
```
 -Pconsume-incrementals
 -Pmight-produce-incrementals
-Dchangelist.format=%d.v%s
```

pom.xml
```
   <artifactId>some-library-wrapper</artifactId>
   <version>${revision}.${changelist}</version>
   <packaging>hpi</packaging>
   <properties>
     <revision>1</revision>
     <changelist>999999-SNAPSHOT</changelist>
     <jenkins.version>2.176.4</jenkins.version>
```

Reply:
Carelessly applied, that changes both your artifact ID and Jenkins baseline.

Reply:
well, I give some credit to the Jenkins plugin maintainer to understanding the `pom.xml` and the `maven.config` files

Reply:
Explain how we have these CD-"enabled" plugins then, where maintainers evidently failed to follow these instructions:

```
console-column-plugin:2.0.1-rc59.8fe210d5e9b_3
jobConfigHistory:2.31-rc1107.2354f08725a_8
leanix-microservice-intelligence:1.0.11-rc201.335c08161b66
minio:1.3.3-rc93.9e92f846d4cf
purge-build-queue-plugin:2.0.0-rc11.9c20b_30c9572
remoting-opentelemetry:1.0-rc89.d67d14d05962
skip-cron-rebuild:1.1-rc20.50497e5a8a3a
talend:1.4-rc43.dbb2c0671f67